### PR TITLE
Fastapi tutorial

### PR DESCRIPTION
**Pull request by aimurphy**
https://64c05164aa64f5006ebba9fe--harness-developer.netlify.app/tutorials/ci-pipelines/test/fastapi

### CODE REVIEW

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+
+## Prerequisites
+
+You need the following for this tutorial:
+
+* Knowledge of Python and FastAPI.
+* Git and GitHub basics.
+* [A GitHub Account](https://github.com/join).
+* [A Harness Account](https://app.harness.io/).
+
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:33]:
```suggestion

* Knowledge of Python, FastAPI, Git, and GitHub.
* [A GitHub account](https://github.com/join).
* [A Harness account](https://app.harness.io/).

```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+1. In your Harness project, go to the CI module.
+2. Select **Pipelines**, and then select **Create a Pipeline**.
+3. Enter a **Name** and select **Start**.
+4. Add a **Build** (`CI`) stage.
+
+### Add the Build stage and infrastructure
+
+```mdx-code-block
+<Tabs>
+  <TabItem value="Visual" label="Visual">
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:123]:
```suggestion
  <TabItem value="Visual" label="Visual" default>
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+
+1. Select **Add Stage**, and select the **Build** stage.
+2. Enter a **Stage Name**, such as `Test FastAPI`.
+3. For **Connector**, select the GitHub connector you created earlier in [Prepare the codebase](#prepare-the-codebase).
+4. Select **Set Up Stage**.
+5. In the **Build** stage, select the **Infrastructure** tab, and [set up your build infrastructure](/docs/category/set-up-build-infrastructure).
+
+```mdx-code-block
+  </TabItem>
+  <TabItem value="YAML" label="YAML" default>
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:134]:
```suggestion
  <TabItem value="YAML" label="YAML">
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+
+```mdx-code-block
+<Tabs>
+  <TabItem value="hosted" label="Harness Cloud" default>
+```
+
+```yaml
+              - step:
+                  type: Run
+                  name: Run Pytest
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:247]:
```suggestion
                  name: Pytest
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+                  spec:
+                    shell: Sh
+                    command: |-
+                      sudo apt-get update && sudo apt-get install -y python3-dev && sudo apt-get install default-libmysqlclient-dev
+                      pip install --cache-dir .pip_cache -r requirements.txt
+                    envVariables:
+                      PIP_CACHE_DIR: /root/.cache
+              - step:
+                  type: Run
+                  name: Run Pytest
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:371]:
```suggestion
                  name: Pytest
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+
+GitHub webhooks are usually automatically created in the target repo. If automatic registration fails, you must manually copy the webhook URL and add it to your repo webhooks. For instructions on manual webhook registration, go to [Register the webhook in the Git provider](/docs/platform/Triggers/triggering-pipelines#register-the-webhook-in-the-git-provider).
+
+## Run the pipeline
+
+To test the Git event trigger and run the pipeline, go to your FastAPI repo fork, make a change, and commit and push it to `main`. For this tutorial, you could commit directly to `main`, but in a real world development situation, you would want to create and merge a PR.
+
+Upon pushing to `main` (either directly or by merging a PR), the trigger should start your pipeline within a few moments. While the build runs, you can view the logs and monitor build activity on the [Build details page](docs/continuous-integration/use-ci/viewing-builds).
+
+After the pytest step, if the step succeeded, you find logs indicating that the `output-test.xml` file was generated, and you can [view the test results](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests) on the Tests tab.
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:313]:
```suggestion
After the pytest step runs, you can find logs indicating that the `output-test.xml` file was generated, and you can [view the test results](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests) on the Tests tab.
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+                        paths:
+                          - output-test.xml
+```
+
+```mdx-code-block
+</TabItem>
+<TabItem value="selfhosted" label="Self-hosted">
+```
+
+This pipeline uses a [Kubernetes cluster build infrastructure](/docs/category/set-up-kubernetes-cluster-build-infrastructures).
```
Comment (aimurphy) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:389]:
```suggestion
This example uses a [Kubernetes cluster build infrastructure](/docs/category/set-up-kubernetes-cluster-build-infrastructures).
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+
+<CISignupTip />
+
+## Prepare the codebase
+
+1. Fork the [fastapi-harness-sample repository](https://github.com/harness-community/fastapi-harness-sample) into your GitHub account.
+2. In Harness, [create a project](/docs/platform/organizations-and-projects/create-an-organization#create-a-project) or select an existing project, and then go to the CI module.
+3. Select **Connectors** under **Project Setup**.
+4. Select **New Connector**, and select **GitHub** under **Code Repositories**.
+5. Configure the [GitHub connector settings](/docs/platform/Connectors/Code-Repositories/ref-source-repo-provider/git-hub-connector-settings-reference) so that the connector can access your fork of the sample repo.
```
Comment (michaelcretzman) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:46]: you could probably provide more guidance here.

File: tutorials/ci-pipelines/test/ci-fastapi-test.md [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+
+To [view test reports in Harness](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests/), test results must be in JUnit XML format, and the `reports` specification must be included.
+
+## Add the trigger
+
+You can run this pipeline manually as it is, or you can add a trigger to automatically run these tests whenever the codebase changes. To do this, add a [Git event trigger](docs/platform/Triggers/triggering-pipelines) that listens for an event on a specific branch of your FastAPI repo fork.
+
+For this tutorial, you'll create a trigger that listens for pushes to the `main` branch.
+
+1. In Harness, select **Triggers** in the Pipeline Studio header, and then select **Add New Trigger**.
```
Comment (michaelcretzman) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:296]:
```suggestion
1. In Harness, in the same pipeline, select **Triggers** in the Pipeline Studio header, and then select **Add New Trigger**.
```

File: tutorials/ci-pipelines/test/ci-fastapi-test.md (hunk 6 above)
Comment (michaelcretzman) [inline tutorials/ci-pipelines/test/ci-fastapi-test.md:313]:
```suggestion
After the pytest step, if the step succeeded, you'll find logs indicating that the `output-test.xml` file was generated, and you can [view the test results](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests) on the **Tests** tab.
```

**Review by michaelcretzman — APPROVED** [review]
Approved with suggestions.

File: tutorials/ci-pipelines/test/ci-fastapi-test.md (hunk 8 above)
Comment (aimurphy) [reply tutorials/ci-pipelines/test/ci-fastapi-test.md:45]: CI PMs are pushing for the tutorials to look quick/easy, thus why the connector setup instructions are minimal here.

**Pull request merged**